crates/env/src/types.rs

```diff
@@ -106,6 +106,15 @@ pub trait AccountIdGuard {}
 /// used in the [`DefaultEnvironment`].
 impl AccountIdGuard for AccountId {}
 
+#[cfg(feature = "std")]
+pub trait CodecAsType: scale_decode::DecodeAsType + scale_encode::EncodeAsType {}
+#[cfg(feature = "std")]
+impl<T: scale_decode::DecodeAsType + scale_encode::EncodeAsType> CodecAsType for T {}
+#[cfg(not(feature = "std"))]
+pub trait CodecAsType {}
+#[cfg(not(feature = "std"))]
+impl<T> CodecAsType for T {}
```

Comment:
This crate already uses `cfg_if!` so would be good to use that here.

Reply:
sure, I missed that - refactored